src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -76,6 +79,16 @@
     boolean setLocation(Location location);
 
     /**
+     * Moves the entity by the specified
+     *
+     * @param location The location to set
+     * @param rotation The rotation to set
+     * @param relativePositions The coordinates to set relatively
+     * @return True if the teleport was successful
+     */
+    boolean setLocationAndRotation(Location location, Vector2f rotation, EnumSet<RelativePositions> relativePositions);
```

Comment:
Network protocol on the teleport allows setting relative mode for each component of location/pitch/yaw separately.

E.g., /teleport 0 ~ ~3 sends you to (0, y, z+3), and would be Location(0, 0, 3), Rotation(0, 0), EnumSet.of(RelativePositions.Y, RelativePositions.Z, RelativePositions.PITCH, RelativePositions.YAW)